web/src/main/java/org/springframework/security/web/authentication/AuthenticationEntryPointRethrowingFailureHandler.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.authentication;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.security.authentication.AuthenticationServiceException;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.util.Assert;
+
+/**
+ * Adapts a {@link AuthenticationEntryPoint} into a {@link AuthenticationFailureHandler}.
+ * When the failure is an {@link AuthenticationServiceException}, it re-throws, to produce
+ * an HTTP 500 error.
+ *
+ * @author Daniel Garnier-Moiroux
+ * @since 5.8
+ */
+public class AuthenticationEntryPointRethrowingFailureHandler implements AuthenticationFailureHandler {
```

Comment:
I don't like the names since, if the other implementation didn't already exist, I don't believe this one would be named this way. The fact that it rethrows `AuthenticationServiceException` doesn't seem such a defining characteristic so as to embed it in the name. Also, "rethrowing" isn't very descriptive in the end.

I think `AuthenticationEntryPointFailureHandlerAdapter` or `AuthenticationEntryPointAdaptingFailureHandler` would be better. There is precedent for `XXXAdapter` classes in Spring Security, like `ReactiveJwtAuthenticationConverterAdapter`.

Reply:
Ah, thank you, I like names better as well.